components/Card/ActivityLogCard.js

```diff
@@ -1,64 +1,54 @@
-import { Pressable, Text, View } from "react-native";
-import { ScrollView } from "react-native-gesture-handler";
+import React, { useState, useEffect } from "react";
+import { Text, View } from "react-native";
 import { useNavigation } from "@react-navigation/native";
 import { useIsFocused } from "@react-navigation/native";
 
-import ExpandableCard from "components/Card/ExpandableCard";
-import PrefetchCacheRecord from "components/PrefetchCacheRecord";
+import Card from "components/Card/Card";
 
 import useI18N from "hooks/use-i18n";
 import useActivityLog from "hooks/use-activity-log";
-import useBottomSheet from "hooks/use-bottom-sheet";
+
 import useStyles from "hooks/use-styles";
-import useType from "hooks/use-type";
 
 import { ScreenConstants } from "constants";
 
 import { localStyles } from "./ActivityLogCard.styles";
+import RenderActivityLog from "components/RenderActivityLog";
 
 const ActivityLogCard = ({ preview, refreshing }) => {
   // NOTE: invoking this hook causes the desired re-render onBack()
   useIsFocused();
 
+  const [accordionState, setAccordionState] = useState([]);
   const navigation = useNavigation();
   const { styles, globalStyles } = useStyles(localStyles);
-  const { getTabScreenFromType } = useType();
-  const { collapse } = useBottomSheet();
   const { i18n, numberFormat } = useI18N();
 
+  useEffect(() => {
+    if (groupedActivityLog) {
+      setAccordionState(
+        new Array(Object.keys(groupedActivityLog).length).fill(false)
+      );
+    }
+  }, [groupedActivityLog]);
+
+  const handleAccordionChange = (groupIndex) => {
+    const updatedAccordionState = accordionState.map((item, index) =>
+      index === groupIndex ? !item : item
+    );
+    setAccordionState(updatedAccordionState);
+  };
+
   // TODO: FilterList
-  const renderActivityLog = (log, idx) => {
-    const _key = `${log.object_id}-${idx}`;
+  const renderActivityLog = (logs, idx) => {
     return (
-      <View
-        key={_key}
-        style={[globalStyles.columnContainer, styles.activityView]}
-      >
-        <Pressable
-          onPress={() => {
-            const type = log?.post_type;
-            const tabScreen = getTabScreenFromType(type);
-            navigation.jumpTo(tabScreen, {
-              screen: ScreenConstants.DETAILS,
-              id: log?.object_id,
-              name: log?.object_name,
-              type,
-            });
-            collapse();
-          }}
-        >
-          <Text style={styles.activityLink}>{log?.object_name}</Text>
-        </Pressable>
-        <Text style={styles.activityText}>{log?.object_note}</Text>
-        {
-          // Prefetch any posts in the ActivityLog so that the records
-          // are available if the user goes OFFLINE.
-        }
-        {log?.object_id && log?.post_type && (
-          <PrefetchCacheRecord id={log.object_id} type={log.post_type} />
-        )}
-      </View>
+      <RenderActivityLog
+        key={logs[0]}
```

Comment:
Do we need a default value here to prevent a crash in case logs is undefined? 

eg, `key={logs?.[0] ?? 0`

Reply:
Hi @zdmc23. I have changed it to `key={logs?.[0] ?? Math.random()}`.